daemon/stats.go

```diff
@@ -58,14 +59,64 @@ func (daemon *Daemon) ContainerStats(name string, config *ContainerStatsConfig)
 				return nil
 			}
 
-			s := getStat(v)
+			statsJSON := getStatJSON(v)
+			if version.LessThan("1.21") {
```

Comment:
can we have booleans in `ContainerStatsConfig` saying which format to use instead of leaking API version here in daemon?